Common/UI/UIScreen.cpp

```diff
@@ -88,6 +95,11 @@ void UIScreen::preRender() {
 	draw->BindFramebufferAsRenderTarget(nullptr, { RPAction::CLEAR, RPAction::CLEAR, RPAction::CLEAR, 0xFF000000 }, "UI");
 	screenManager()->getUIContext()->BeginFrame();
 
+#ifdef OPENXR
+	GLuint fbo = VR_Framebuffer(VR_GetEngine(), 0);
+	glBindFramebuffer(GL_FRAMEBUFFER, fbo);
+#endif
```

Comment:
I would much prefer that we made this and others like it a runtime check rather than an ifdef (even if set by a define during build) to avoid conditional compilation as much as possible, due to how easy it is to cause various kinds of build breakages. Though, I don't know how easy it is for the OpenXR headers to pass build on all platforms...

Reply:
I am quite sure it would fail on iOS/OSX. I used the defines to ensure I do not break anything (especially as the tests don't run automatically for me).

Reply:
We currently build with VK headers and decide not to use them at runtime, for example, so sometimes this is possible. But I won't require it to be done before merge.

I have now approved you for CI on this project.

Reply:
Perfect, thanks

Reply:
Note: the approval is temporary until something is merged.

I'd suggest opening a separate pull with some small subset of changes, like maybe adding the DEVICE_TYPE_VR stuff (but nothing that would ever trigger it or the OPENXR defines, yet.)  If that's merged, the CI will run henceforth on this PR (which you would need to rebase, though.)

-[Unknown]